fastai2/learner.py

```diff
@@ -328,12 +328,14 @@ def get_preds(self, ds_idx=1, dl=None, with_input=False, with_decoded=False, wit
                 if with_decoded: res.insert(pred_i+2, getattr(self.loss_func, 'decodes', noop)(res[pred_i]))
             return tuple(res)
 
-    def predict(self, item, rm_type_tfms=None):
-        dl = self.dls.test_dl([item], rm_type_tfms=rm_type_tfms)
-        inp,preds,_,dec_preds = self.get_preds(dl=dl, with_input=True, with_decoded=True)
-        i = getattr(self.dls, 'n_inp', -1)
-        full_dec = self.dls.decode_batch((*tuplify(inp),*tuplify(dec_preds)))[0][i:]
-        return detuplify(full_dec),dec_preds[0],preds[0]
+    def predict(learn, item, rm_type_tfms=None, with_input=False):
+        dl = learn.dls.test_dl([item], rm_type_tfms=rm_type_tfms)
+        inp,preds,_,dec_preds = learn.get_preds(dl=dl, with_input=True, with_decoded=True)
+        i = getattr(learn.dls, 'n_inp', -1) - 1
+        dec_inp, full_dec = map(detuplify, learn.dls.decode_batch((*tuplify(inp),*tuplify(dec_preds)))[0][i:])
```

Comment:
Putting the -1 on i doesn't necessarily give you the input (your model could have several inputs, like in tabular). i is the index to split between input and targets.
Those two lines should be rewritten with 
- no -1 for i
- dec_inp being everything up to i (not included)
- full_dec being everything from i and after (and also be renamed dec_targ maybe?)